### PR TITLE
Allowing testing of empty string for selector

### PR DESCRIPTION
**Pull request by irv-fs**
This pull request allows string values to be quoted with `'` or `"` values.  To be backwards compatible, the quotes are optional.  In the code I simply look for them in string values and strip them off if they are there.  Tests were updated to try with and without quotes, as well as testing for empty string.  Empty string can be tested by doing ?(attribute='') in the selector.

### CODE REVIEW

File: src/selector.ts [hunk 1]
```diff
@@ -393,14 +393,25 @@ class PathElement {
...
+        case 'string': {
+          let realString = opProp.value;
+          // look for quoted values, and strip the quotes off if they are there
+          if (opProp.value) {
+            const firstCharacter = realString.charAt(0);
+            const lastCharacter = realString.charAt(realString.length - 1);
+            if ((firstCharacter === '"' && lastCharacter === '"')
+              || (firstCharacter === "'" && lastCharacter === "'")) {
+              realString = realString.substring(1, realString.length - 1);
+            }
```
Comment (irv-fs) [inline src/selector.ts:405]: String the matching `'` or `"` characters off the string and set value into realString.  Then use that everywhere else in the string comparison.  Makes it backwards compatible.

File: test/selector.spec.ts [hunk 2]
```diff
@@ -7,6 +7,7 @@ import { select, validate, Path } from '../src/selector';
 const testData = {
```
Comment (irv-fs) [inline test/selector.spec.ts:7]: Most of this is just taking existing tests and adding in tests for quoting them.  Also empty string tests added.

**Review by ndonze-fs — APPROVED** [review]
This mostly LGTM. The one question I have is if we should be testing that a value with a single `'` or a single `"` behaves as expected. This would be an invalid selector; does it behave the same as other invalid selectors / selector parsing issues?

**Comment by ndonze-fs** [comment]
@irv-fs Once we get this merged I think we'll also want to look at a patch version bump for the library and to document this change in `CHANGELOG`. Let me know if you want any help with that. We could include the version bump and `CHANGELOG` entry in this PR or under a separate PR.

**Review by ndonze-fs — APPROVED** [review]
LGTM

**Pull request merged**